staging/src/k8s.io/apiserver/pkg/server/networkproxy/config_test.go

```diff
@@ -0,0 +1,221 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package networkproxy
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/apis/apiserver"
+)
+
+func TestReadNetworkProxyConfiguration(t *testing.T) {
+	e1 := "unable to read network proxy configuration from \"test-network-proxy-config-absent\" [open test-network-proxy-config-absent: no such file or directory]"
+	e2 := "invalid service configuration object \"\""
+	e3 := "invalid service configuration object \"DaemonSet\""
+
+	testcases := []struct {
+		name           string
+		contents       string
+		createFile     bool
+		expectedResult *apiserver.ConnectivityServiceConfiguration
+		expectedError  *string
+	}{
+		{
+			name:           "empty",
+			createFile:     true,
+			contents:       ``,
+			expectedResult: nil,
+			expectedError:  &e2,
```

Comment:
defining a strptr function and putting the string here directly is more readable...

Reply:
Good suggestion. Fixed.